src/cpp/rtps/transport/shared_mem/SharedMemGlobal.hpp

```diff
@@ -689,6 +707,63 @@ class SharedMemGlobal
             }
         }
 
+        /**
+         * @brief Look for a listener processing a buffer and return the buffer descriptor being processing
+         * 
+         * Iterates over all the listeners, and upon founding the first one that is marked as \c is_processing:
+         *  - Marks it as not processing
+         *  - Copies the buffer descriptor that was being processed in the provided buffer
+         *  - Finishes iterating
+         * 
+         * The intention of this method is to locate all buffer descriptors that are being processed
+         * and apply necessary operations on these buffers when the port is found to be zombie.
```

Comment:
Does this mean that a descriptor may be returned and the listener be marked as not processing even if the port is not zombie? If so, I think this is not the correct behavior.

Looking at this carefully, it seems that a precondition of this method is that the port is zombie. Please document that.

Reply:
yes, that is a precondition. Documented.